python/cog/server/redis_queue.py

```diff
@@ -215,7 +215,10 @@ def handle_message(self, response_queue, message, cleanup_functions):
         with self.capture_log(self.STAGE_RUN, prediction_id), timeout(
             seconds=self.predict_timeout
         ):
-            return_value = self.predictor.predict(**input_obj.dict())
+            try:
+                return_value = self.predictor.predict(**input_obj.dict())
+            finally:
+                input_obj.cleanup()
```

Comment:
This isn't actually tested. We don't have any unit tests so this is very hard to test. It's a very simple copy & paste of the HTTP implementation which is tested.